mesh/meshdb.go

```diff
@@ -1069,3 +1076,23 @@ func (m *DB) cacheWarmUpFromTo(from types.LayerID, to types.LayerID) error {
 
 	return nil
 }
+
+// NewBlockFetcherDB returns reference to a BlockFetcherDB instance.
+func NewBlockFetcherDB(mdb *DB) *BlockFetcherDB {
+	return &BlockFetcherDB{mdb: mdb}
+}
+
+// BlockFetcherDB implements API that allows fetcher to get a block from a remote database.
+type BlockFetcherDB struct {
+	mdb *DB
+}
+
+// Get types.Block encoded in byte using hash.
+func (db *BlockFetcherDB) Get(hash []byte) ([]byte, error) {
+	id := types.BlockID(types.BytesToHash(hash).ToHash20())
+	blk, err := db.mdb.GetBlock(id)
```

Comment:
Combine with the following line

Reply:
did you mean to combine id initialization with this line?

Reply:
I mean move the if... conditional up one line and combine it with this line (line 1093)